src/validation.cpp

```diff
@@ -1157,19 +1158,23 @@ CAmount GetBlockSubsidyInner(int nPrevBits, int nPrevHeight, const Consensus::Pa
         nSubsidy *= consensusParams.nHighSubsidyFactor;
     }
 
+    CAmount nSuperblockPart{};
     // Hard fork to reduce the block reward by 10 extra percent (allowing budget/superblocks)
-    CAmount nSuperblockPart = (nPrevHeight > consensusParams.nBudgetPaymentsStartBlock) ? nSubsidy/10 : 0;
-
+    if (nPrevHeight > consensusParams.nBudgetPaymentsStartBlock) {
+        nSuperblockPart = nSubsidy / 10;
+        // Once MNRewardReallocated is active, the treasury is 20% instead of 10%
+        if (fMNRewardReallocated) nSuperblockPart *= 2;
```

Comment:
```suggestion
        // Once MNRewardReallocated is active, the treasury is 20% instead of 10%
        nSuperblockPart = nSubsidy / (fMNRewardReallocated ? 5 : 10);
```